src/physics/em/detail/MuBremsstrahlungInteractor.i.hh

```diff
@@ -106,87 +104,93 @@ CELER_FUNCTION Interaction MuBremsstrahlungInteractor::operator()(Engine& rng)
 /*!
  * Sample cosine of the angle between incident and secondary particles.
  */
-
 template<class Engine>
 CELER_FUNCTION real_type MuBremsstrahlungInteractor::sample_cos_theta(
     real_type gamma_energy, Engine& rng) const
 {
-    const real_type gamma = 1.0 + inc_energy_.value() / inc_mass_.value();
+    const real_type gamma = particle_.lorentz_factor();
     const real_type r_max
-        = gamma * constants::pi * 0.5
-          * min(1.0, gamma * inc_mass_.value() / gamma_energy - 1);
+        = gamma * constants::pi * real_type(0.5)
+          * min(real_type(1.0),
+                gamma * particle_.mass().value() / gamma_energy - 1);
     const real_type r_max_sq = ipow<2>(r_max);
 
-    UniformRealDistribution<real_type> xi;
-    const real_type                    a = xi(rng) * r_max_sq / (1 + r_max_sq);
+    const real_type a = generate_canonical(rng) * r_max_sq / (1 + r_max_sq);
 
-    return std::cos(std::sqrt(a / (1.0 - a)) / gamma);
+    return std::cos(std::sqrt(a / (1 - a)) / gamma);
 }
 
 //---------------------------------------------------------------------------//
 
 CELER_FUNCTION real_type MuBremsstrahlungInteractor::differential_cross_section(
     real_type gamma_energy) const
 {
-    real_type dxsection = 0.0;
+    real_type dxsection = 0;
 
-    if (gamma_energy >= inc_energy_.value())
+    if (gamma_energy >= particle_.energy().value())
     {
         return dxsection;
     }
 
     const int       atomic_number    = element_.atomic_number();
     const real_type atomic_mass      = element_.atomic_mass().value();
     const real_type sqrt_e           = std::sqrt(constants::euler);
-    const real_type inc_total_energy = inc_mass_.value() + inc_energy_.value();
+    const real_type inc_total_energy = particle_.mass().value()
+                                       + particle_.energy().value();
     const real_type rel_energy_transfer = gamma_energy / inc_total_energy;
-    const real_type inc_mass_sq         = ipow<2>(inc_mass_.value());
-    const real_type delta = 0.5 * inc_mass_sq * rel_energy_transfer
+    const real_type inc_mass_sq         = ipow<2>(particle_.mass().value());
+    const real_type delta = real_type(0.5) * inc_mass_sq * rel_energy_transfer
                             / (inc_total_energy - gamma_energy);
 
     real_type       d_n_prime, b, b1;
-    const real_type d_n = 1.54 * std::pow(atomic_mass, 0.27);
+    const real_type d_n = real_type(1.54)
+                          * std::pow(atomic_mass, real_type(0.27));
 
     if (atomic_number == 1)
     {
         d_n_prime = d_n;
-        b         = 202.4;
-        b1        = 446.0;
+        b         = real_type(202.4);
+        b1        = 446;
     }
     else
     {
-        d_n_prime = d_n / std::pow(d_n, 1.0 / atomic_number);
-        b         = 183.0;
-        b1        = 1429.0;
+        d_n_prime = d_n / std::pow(d_n, 1 / atomic_number);
```

Comment:
An integer division snuck in (`1 / atomic_number` will always be 0).

Reply:
Also, I see the usage of both `std::pow()` and `ipow<n>()`. Aside from consistency, do we have to worry if they're used interchangeably?

Reply:
There it is!! Thanks amanda, I was poring over the changeset for 15 minutes to find what changed before seeing your comment. Thanks!

You could also simplify the expression slightly while you're at it,
```suggestion
        d_n_prime = std::pow(d_n, 1 - real_type(1) / atomic_number);
```

Reply:
> Also, I see the usage of both `std::pow()` and `ipow<n>()`. Aside from consistency, do we have to worry if they're used interchangeably?

`ipow` should always be used for known integer `n`; otherwise you have to use `std::pow` (or some other runtime expression).